external-crates/move/crates/move-analyzer/trace-adapter/src/runtime.ts

```diff
@@ -793,23 +955,30 @@ export class Runtime extends EventEmitter {
     public toString(): string {
         let res = 'current frame stack:\n';
         for (const frame of this.frameStack.frames) {
-            const fileName = path.basename(frame.file);
+            const fileName = frame.showDisassembly ?
+                path.basename(frame.bcodeFilePath!) :
+                path.basename(frame.srcFilePath);
+            const line = frame.showDisassembly ? frame.bcodeLine : frame.srcLine;
             res += this.singleTab
                 + 'function: '
                 + frame.name
                 + ' ('
                 + fileName
                 + ':'
-                + frame.line
+                + line
                 + ')\n';
             for (let i = 0; i < frame.locals.length; i++) {
                 res += this.singleTab + this.singleTab + 'scope ' + i + ' :\n';
                 for (let j = 0; j < frame.locals[i].length; j++) {
                     const local = frame.locals[i][j];
-                    if (local) {
-                        res += this.varToString(this.singleTab
-                            + this.singleTab
-                            + this.singleTab, local) + '\n';
+                    if (local && (frame.showDisassembly ||
+                        (!local.info.name.includes('%') && !local.info.internalName.includes('%')))) {
```

Comment:
Nit: I see this same type of predicate used elsewhere (and I'm guessing used in other places as well), so might be nice to pull this out into a separate function that can be shared e.g., `is_generated_local` or something like that.

Reply:
Done!